### PR TITLE
Use iota enums for states in InMemBackend

### PR DESCRIPTION
**Pull request by madhuravi**

### CODE REVIEW

File: modules/task/backend.go [hunk 1]
```diff
@@ -30,6 +30,12 @@ import (
 
 var gobEncoding = &GobEncoding{}
 
+const (
+	_initialized = iota
```
Comment (sectioneight) [inline modules/task/backend.go]: hmm i think breaking the underscore rule here might be warranted. this is hard to look at

Comment (madhuravi) [inline modules/task/backend.go]: Either one is ok but we should be consistent across our repo. Let me know if you want to start following no underscores for constants.

Comment (glibsm) [inline modules/task/backend.go]: i think the convention for the first one is typically `unknown` or something like that, and the three valid states are listed below. I might be wrong though

Comment (madhuravi) [inline modules/task/backend.go]: @glibsm I don't see that in examples online. Let me know if you have examples. I personally don't like unknown, the name itself does not indicate what it is and is not meaningful here.

**Pull request merged**